Changes to indent rule for empty blocks in eslint@1

Previously we had code similar to this dummy code below and it looks and linted fine in eslint@0.24.x. Since moving to eslint@1 (awesome work, btw!) it seems like we're getting a few rogue errors like:

> "Expected indentation of 4 characters but found 6"

---

### dummy.js
```js
/*eslint-env node */

var p = {};

var obj = {
  foo: function () {
    return new p()
      .then(function (ok) {
        return ok;
      }, function () {
        // ignore things
      });
  }
};
console.log(obj);
```

### .eslintrc
```json
{
  "extends": "eslint:recommended",
  "rules": {
    "handle-callback-err": 0,
    "no-console": 0,
    "strict": [2, "never"],
    "indent": [2, 2]
  }
}
```

### Output:
```
➜  fxa-esl1 git:(devdeps44) ✗ npm run lint

> fxa-content-server@0.43.0 lint /Users/pdehaan/dev/tmp/del/fxa-esl1
> eslint dummy.js


dummy.js
  12:8  error  Expected indentation of 4 characters but found 6  indent

✖ 1 problem (1 error, 0 warnings)
```

### Workaround:
If I'm understanding this correctly, this seems to be because my "ignore things" function is an empty block and doesn't have any statements. If I change that to `return;` then the `eslint dummy.js` succeeds.

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -238,6 +238,19 @@ module.exports = function(context) {
             indent = getNodeIndent(calleeNode);
         }
 
+        // If part of a multi-line chain call, indent one more to differentiate:
+        // var a = new P()
+        //     .done(function (result) {
+        //         doUsefulStuff(result);
+        //     });
+        var calleeLoc = calleeNode.type === "CallExpression" && calleeNode.callee.loc;
+        if (calleeLoc &&
+            calleeNode.loc.start.line < calleeLoc.end.line &&
+            calleeLoc.end.line < node.loc.start.line
+        ) {
+            indent += indentSize;
+        }
+
         indent += indentSize;
         // If function content is not empty
         if (node.body.length > 0) {

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -718,6 +718,28 @@ ruleTester.run("indent", rule, {
                 "}\n",
             ecmaFeatures: { modules: true, defaultParams: true },
             options: [2]
+        },
+        {
+            code:
+                "var obj = {\n" +
+                "  foo: function () {\n" +
+                "    return new p()\n" +
+                "      .then(function (ok) {\n" +
+                "        return ok;\n" +
+                "      }, function () {\n" +
+                "        // ignore things\n" +
+                "      });\n" +
+                "  }\n" +
+                "};\n",
+            options: [2]
+        },
+        {
+            code:
+                "a.b()\n" +
+                "  .c(function(){\n" +
+                "    var a;\n" +
+                "  }).d.e;\n",
+            options: [2]
         }
     ],
     invalid: [

